build.sbt

```diff
@@ -88,123 +76,155 @@ lazy val root = (project in file("."))
 
 lazy val allOld = (project in file("allOld"))
   .dependsOn(
-    core % "compile->compile;provided->provided",
-    scalatest % "compile->provided;provided->provided",
-    scalatestSelenium % "compile->provided;provided->provided",
-    moduleMysql % "compile->provided;provided->provided",
-    modulePostgres % "compile->provided;provided->provided",
-    moduleCassandra % "compile->provided;provided->provided",
-    moduleKafka % "compile->provided;provided->provided",
-    moduleVault % "compile->provided;provided->provided"
+    core,
+    scalatest,
+    scalatestSelenium,
+    moduleMysql,
+    modulePostgres,
+    moduleCassandra,
+    moduleKafka,
+    moduleVault,
   )
   .settings(commonSettings: _*)
   .settings(
     name := "testcontainers-scala",
-    libraryDependencies ++= PROVIDED(
-      "org.scalatest" %% "scalatest" % scalaTestVersion
-    )
+    libraryDependencies ++= Dependencies.allOld.value ++ (
+      Dependencies.scalatestSelenium.value ++
+      Dependencies.moduleMysql.value ++
+      Dependencies.modulePostgres.value ++
+      Dependencies.moduleCassandra.value ++
+      Dependencies.moduleKafka.value ++
+      Dependencies.moduleVault.value
+    ).collect {
+      case module if module.configurations.isEmpty =>
+        module.withConfigurations(Some("provided"))
+    },
+    pomPostProcess := { root =>
```

Comment:
Can we hide this method in a separate file in `project/` (so that build.sbt is a bit more readable)?

Reply:
I tried but it looks like not an easy quest. I use module names and stuff from `Dependencies` object inside. And to access them I need to define this function as a `Def.something`. I played a bit with it and failed.

I'm absolutely sure that this is doable, but it's not easy for me.

Reply:
I hope this solution with pom.xml is temporary and we eventually will delete all this weird code.